irony-cdb-json.el

```diff
@@ -48,6 +48,53 @@ source directory.")
     (get-compile-options (irony-cdb-json--get-compile-options))))
 
 ;;;###autoload
+(defun irony-cdb-json--find-common-root-directory (compile-commands-path)
+  "Return the common root directory for all source files listed in a cdb.
+
+Reads compilation database at COMPILE-COMMANDS-PATH, extracts all source file paths
+and returns the deepest directory which contains them all.
+
+Example:
+If cdb contains entries for \"/foo/bar/baz\" \"/foo/bar/baaz\" and \"/foo/bozze\"
+will return \"/foo\""
+  (let* ((files (mapcar 'car (irony-cdb-json--load-db (expand-file-name  compile-commands-path))))
+         (dirs (delete-dups (mapcar (lambda (x) (file-name-directory (expand-file-name x))) files )))
+         (dir-components (mapcar (lambda (dir)
+                                   (cl-remove-if (lambda (v) (string= "" v)) (split-string dir "/")))
+                                    dirs))
+         (minlen (apply 'min (mapcar 'length dir-components)))
+         (common-count (cl-loop for i in (number-sequence 0 minlen)
+                                while (= 1 (length (delete-dups (mapcar (lambda (val) (nth i val)) dir-components))))
+                                finally return i
+                                )))
+    (mapconcat 'file-name-as-directory (cons "/" (-take common-count (car dir-components))) "")
+    ))
+
+;;;###autoload
+(defun irony-cdb-json-add-compile-commands-path-auto-root (compile-commands-path)
```

Comment:
I would change `irony-cdb-json-add-compile-commands-path` to take the arguments in the reverse order.
First the build directory or compilation database.
Then the second argument could be pre-filled with what you guessed with `irony-cdb-json--find-common-root-directory`.
We will have only one function but the user is helped a bit so he can just press `RET` but he is still able to specify a custom build dir (the guess will be wrong on quite a few of my projects).